src/app/features/animals/components/animals-list/animals-list.component.html

```diff
@@ -1,4 +1,60 @@
-<h1>Animals</h1>
+<div class="row">
+  <div class="col">
+    <h1>Animals</h1>
+  </div>
+  <div class="col text-end">
+    <button class="btn btn-success" routerLink="/animals/add">Add new</button>
+  </div>
+</div>
 
-<!-- TODO: Add table with animals -->
-<p>Coming soon...</p>
+<ul class="nav nav-tabs mb-4">
+  <li class="nav-item">
+    <a class="nav-link" (click)="changeActiveTab('table')" [class.active]="activeTab === 'table'">Table</a>
+  </li>
+  <li class="nav-item">
+    <a class="nav-link" (click)="changeActiveTab('cards')" [class.active]="activeTab === 'cards'">Cards</a>
+  </li>
+</ul>
+
+<div>
+  <ng-container *ngIf="activeTab === 'table'">
+    <table class="table table-striped align-middle">
+      <thead>
+      <tr>
+        <th scope="col">#</th>
+        <th scope="col">Name</th>
+        <th scope="col">Type</th>
+        <th scope="col" class="text-end">Actions</th>
+      </tr>
+      </thead>
+      <tbody>
+      <tr *ngFor="let animal of animalsService.animals; let i = index">
+        <th scope="row">{{ i + 1 }}</th>
+        <td>{{ animal.name }}</td>
+        <td>{{ animal.type }}</td>
+        <td class="text-end">
+          <button type="button" class="btn btn-danger btn-sm" (click)="removeAnimal(animal)">Remove</button>
+        </td>
+      </tr>
+      </tbody>
+    </table>
+  </ng-container>
+
+  <ng-container *ngIf="activeTab === 'cards'">
+    <div class="row">
+      <div *ngFor="let animal of animalsService.animals; let i = index" class="col-3 mb-4">
+        <div class="card" [class.border-danger]="animal.type === 'Wild'"
+             [class.border-success]="animal.type === 'Domestic'">
+          <img src="{{ animal.imageUrl }}" class="card-img-top" alt="animal" style="max-height: 200px;">
```

Comment:
Niektoré zvieratá nemajú nastavený žiadny obrázok a namiesto neho by sa mal použiť obrázok `placeholder.jpg` z priečinka `public`.

![image](https://github.com/Viktor-36/angular-code-review-demo/assets/57718154/d82b715d-8727-433d-aa9a-1eaab2fb678a)

```suggestion
      <img src="{{ animal.imageUrl ?? '/placeholder.jpg' }}" class="card-img-top" alt="animal" style="max-height: 200px;">
```